### DISCUSSION TITLE
Create routes

### DISCUSSION THREAD

**Discussion by dnl0037** [idom-team/idom#1111] [Question] [answered]
Hi there! I would like to know how to set up routes for the pages of my web app. For instance, how can I configure "/main-page" or "/secondary-page"? Additionally, how can I create a button that redirects to any of these routes ?

**Comment by rmorshea** [comment, marked as answer]
Checkout [`reactpy-router`](https://github.com/reactive-python/reactpy-router).

**Reply by dnl0037** [reply]
thank you very much

**Reply by dnl0037** [reply]
Hey, I've been reading the docs, and there are a few parts that don't seem to work: "Using Queries" and "Using Parameters". I'm getting this error:
| ERROR | Failed to render router_component(2203a718880, routes=(Route(path='/', element={'tagName': 'h1', 'children': ['Home Page 🏠']}, routes=()), Route(path='/user/{id:int}', element=user(2203ad4ac20), routes=()), link(2203a718700, children=('User 123',), to='/user/123')), compiler=<class 'reactpy_router.simple.SimpleResolver'>) 
AttributeError: 'Component' object has no attribute 'routes'

Also, when importing "use_location", I can't do it from "reactpy_router" as the docs say. Instead, I need to import it from "reactpy". 
By the way, if you don't mind, could you give me some advice on how to implement the "handle_click" function here:

```python
@component
def first_page():
    def handle_click(event):
        "go to second page"

    return html.div("First Page",
                    html.button({"on_click": handle_click}, "2nd page"))


@component
def second_page():
    return html.h1("Second Page")


@component
def root():
    return simple.router(
        route("/", first_page()),
        route("/second", second_page()),
        route("*", html.h1("Missing Link 🔗‍💥")),

    )
```

**Reply by rmorshea** [reply]
The following renders without error for me:

```python
from reactpy import component, html, run
from reactpy_router import route, simple

@component
def first_page():
    def handle_click(event):
        "Go to second page"

    return html.div("First Page", html.button({"on_click": handle_click}, "2nd page"))

@component
def second_page():
    return html.h1("Second Page")

@component
def root():
    return simple.router(
        route("/", first_page()),
        route("/second", second_page()),
        route("*", html.h1("Missing Link 🔗‍💥")),
    )

run(root)
```

**Reply by rmorshea** [reply]
Ah, I see. [This example](https://reactive-python.github.io/reactpy-router/usage/#route-links) here is wrong. It should instead be:

```python
from reactpy import component, html, run, use_location
from reactpy_router import link, route, simple

@component
def root():
    use_location()
    return simple.router(
        route(
            "/",
            html.div(
                html.h1("Home Page 🏠"),
                link(html.button("About"), to="/about"),
            ),
        ),
        route("/about", html.h1("About Page 📖")),
    )

run(root)
```

**Reply by dnl0037** [reply]
Thanks for that clarification! And, regarding my code, sorry, I didn't explain myself properly. My goal is to make the button redirect me to the second page, but I'm struggling to get it to work. Do you have any advice or tips to help me out?
```python
@component
def first_page():
    use_location()

    def handle_click(event):
        link(to="/second")

    return html.div("First Page", html.button({"on_click": handle_click}, "2nd page"))
```

**Reply by rmorshea** [reply]
```python
return html.div("First Page", link(html.button("2nd page"), to="/second"))
```

**Reply by rmorshea** [reply]
The docs have been updated: https://reactive-python.github.io/reactpy-router/usage/

**Reply by dnl0037** [reply]
aah! excellent! thanks a lot

**Reply by rmorshea** [reply]
If you need more help, see this tutorial as well: https://reactive-python.github.io/reactpy-router/tutorials/simple-app/#nested-routes

**Reply by dnl0037** [reply]
Will do. I appreciate your help.

**Reply by Qwinkyy** [reply]
а как это привязывать к роутам flask?
